sync vs async example v1 (#229)
socotecio/django-socio-grpc: docs/inner-workings/sync-vs-async.rst

.. list-table::
   :widths: 50 50
   :header-rows: 1

   * - Async Class
     - Sync class
   * - ``django_socio_grpc.generics.AsyncCreateService``
     - ``django_socio_grpc.generics.CreateService``
   * - ``django_socio_grpc.generics.AsyncListService``
     - ``django_socio_grpc.generics.ListService``
   * - ``django_socio_grpc.generics.AsyncStreamService``
     - ``django_socio_grpc.generics.StreamService``
   * - ``django_socio_grpc.generics.AsyncRetrieveService``
     - ``django_socio_grpc.generics.RetrieveService``
   * - ``django_socio_grpc.generics.AsyncDestroyService``
     - ``django_socio_grpc.generics.DestroyService``
   * - ``django_socio_grpc.generics.AsyncUpdateService``
     - ``django_socio_grpc.generics.UpdateService``
   * - ``django_socio_grpc.generics.AsyncListCreateService``
     - ``django_socio_grpc.generics.ListCreateService``
   * - ``django_socio_grpc.generics.AsyncReadOnlyModelService``
     - ``django_socio_grpc.generics.ReadOnlyModelService``
   * - ``django_socio_grpc.generics.AsyncModelService``
     - ``django_socio_grpc.generics.class ModelService``

Async ready methods
-------------------

> Since Django 4.1, Django can run ORM queries asynchronously.

Reviewer: ORM is stil synchrone. It's the view that can bu run asynchronously

Check Django `documentation <https://docs.djangoproject.com/en/4.1/topics/async/>`_ for more information.

If you use version of Django < 4.1 or for all django ORM methods not supported,
you can use `asgiref <https://asgi.readthedocs.io/en/latest>`_ to wrap sync methods in async context.

    .. code-block:: python

        from asgiref.sync import sync_to_async
        from buildings.models import Building

        async def a_get(self, pk):
            return await sync_to_async(Building.objects.get)(pk=pk)

In Django Socio gRPC we provide a lot of async ready methods.

#. Services
    * acheck_object_permissions
    * aget_object
    * aget_queryset
    * aget_serializer
#. Serializers
    * asave
    * ais_valid
    * acreate
    * aupdate
    * adata
    * amessage

Sync support
------------

> If you want to use the sync mode, you should know that we will no longer support this mode from version 1.0.0 of Django Socio gRPC.

Reviewer: ```suggestion
If you want to use the sync mode, you should know that we are thinking about droping sync support from version 1.0.0 of Django Socio gRPC.
```
